boardgame-ui/src/pages/LoginPage.vue

```diff
@@ -48,6 +107,49 @@ export default {
         console.log(error);
         this.error = true;
       }
+    },
+
+    async register() {
+      try {
+        await User.signUp(this.user.username, this.user.password, this.user.surname, this.user.name, this.user.email);
+      }
+      catch (error) {
+        console.log(error);
+        this.error = true;
+      }
+    },
+
+    validateBeforeSubmit() {
+      this.$validator.validateAll().then(async (valid) => {
+        if (valid) {
+          if (this.registering) {
+            try {
+              this.register();
```

Comment:
```suggestion
              await this.register();
```
(otherwise, success message displayed even if the server responds with an error)